lib/devices/mtime.ml

```diff
@@ -0,0 +1,15 @@
+open Functoria.DSL
+
+type mtime = job
+
+let mtime = Functoria.job
+let no_mtime = impl "Mirage_runtime" mtime
+
+let impl sublib =
+  let packages =
+    [ package ~min:"5.0.0" ~max:"6.0.0" ~sublibs:[ ""; sublib ] "mirage-mtime" ]
+  in
+  impl ~packages "Mirage_mtime" mtime
+
+let default_mtime = if_impl Key.is_solo5 (impl "solo5") (impl "unix")
```

Comment:
Since both xen and solo5 are using solo5 backend, maybe it easier to check for unix and default to solo5 otherwise.
```suggestion
let default_mtime = if_impl Key.is_unix (impl "unix") (impl "solo5")
```

Reply:
My issue is that `is_xen` exists and is not solo5, so it defaults to unix. The proposed modification gives solo5 for both solo5 and xen. Thinking twice, maybe it should be more explicit.

Reply:
oops, you're right.